yarn-project/simulator/src/avm/avm_machine_state.ts

```diff
@@ -118,40 +117,24 @@ export class AvmMachineState {
     this.output = output;
   }
 
+  public getHalted(): boolean {
+    return this.halted;
+  }
+
+  public getReverted(): boolean {
+    return this.reverted;
+  }
+
+  public getOutput(): Fr[] {
+    return this.output;
+  }
```

Comment:
Consider `public readonly` instead, but I'm ok with getters.

Reply:
I believe that `readonly` means it is essentially immutable (can only be initialized in the constructor), but we want it to be mutable by functions inside the `AvmMachineState`